### PR TITLE
Update ASR RTD Documentation (r1.4)

### PR DESCRIPTION
**Pull request by titu1994**
# Changelog
- Update ASR side documentation (add models, checkpoint to ngc, architecture diagram etc)
- Update doc build script to update mock imports based on requirements

### CODE REVIEW

File: nemo/core/classes/modelPT.py [hunk 1]
```diff
@@ -1112,15 +1061,93 @@ def teardown(self, stage: str):
...
     def prepare_test(self, trainer: 'Trainer') -> bool:
         """
         Helper method to check whether the model can safely be tested
         on a dataset after training (or loading a checkpoint).
 
-        # Usage:
-        trainer = Trainer()
-        if model.prepare_test(trainer):
-            trainer.test(model)
+        ::
```
Comment (titu1994) [inline nemo/core/classes/modelPT.py:1146]: Its a reStructured doc issue, warning pops up otherwise and formatting is a bit off.

File: docs/source/conf.py [hunk 2]
```diff
@@ -38,43 +40,40 @@
...
-    'soundfile',
-    'sentencepiece',
-    'youtokentome',
-    'megatron-lm',
-    'numpy',
-    'dateutil',
-    'wget',
-    'scipy',
-    'pandas',
-    'matplotlib',
```
Comment (tkornuta-nvidia) [inline docs/source/conf.py:62]: ok, this makes me wonder... don't we need to mock up those modules anymore?

File: nemo/collections/nlp/modules/common/megatron/megatron_utils.py [hunk 3]
```diff
@@ -33,7 +33,14 @@
...
 
-MEGATRON_CACHE = os.path.join(_get_torch_home(), "megatron")
+torch_home = _get_torch_home()
+
+if not isinstance(torch_home, str):
+    logging.info("Torch home not found, caching megatron in cwd")
+    torch_home = os.getcwd()
+
+MEGATRON_CACHE = os.path.join(torch_home, "megatron")
+
```
Comment (tkornuta-nvidia) [inline nemo/collections/nlp/modules/common/megatron/megatron_utils.py:43]: hmmm, is this related to this PR?

**Review by okuchaiev — APPROVED** [review]
lgtm

File: docs/source/conf.py (hunk 2 above)
Comment (titu1994) [reply docs/source/conf.py:62]: This mockup is now done below by parsing all of the requirements*.txt from nemo and setting them all up for mock imports

File: nemo/collections/nlp/modules/common/megatron/megatron_utils.py (hunk 3 above)
Comment (titu1994) [reply nemo/collections/nlp/modules/common/megatron/megatron_utils.py:43]: This is to fix NLP docs which break cause of this import. @ericharper fixed it with this workaround.

File: docs/source/conf.py (hunk 2 above)
Comment (tkornuta-nvidia) [reply docs/source/conf.py:62]: oooh, right, I stopped just before this.... this is awesome!!

File: nemo/core/classes/modelPT.py [hunk 4]
```diff
@@ -1112,15 +1061,93 @@ def teardown(self, stage: str):
 
         super().teardown(stage)
 
+    @classmethod
+    def extract_state_dict_from(cls, restore_path: str, save_dir: str, split_by_module: bool = False):
+        """
```
Comment (tkornuta-nvidia) [inline nemo/core/classes/modelPT.py:1066]: Interesting method... somehow I wasn't aware of it before, good that you have polished it a bit ;)

File: nemo/core/classes/modelPT.py (hunk 1 above)
Comment (tkornuta-nvidia) [reply nemo/core/classes/modelPT.py:1146]: is it about the fact that this is a code snippet?

**Review by tkornuta-nvidia — APPROVED** [review]
LGTM

File: nemo/core/classes/modelPT.py (hunk 4 above)
Comment (titu1994) [reply nemo/core/classes/modelPT.py:1066]: In your defense, I wrote this method and then forgot it existed 💯

Comment (tkornuta-nvidia) [reply nemo/core/classes/modelPT.py:1066]: eheheheh, that's the ultimate! ;)

File: nemo/core/classes/modelPT.py (hunk 1 above)
Comment (titu1994) [reply nemo/core/classes/modelPT.py:1146]: Yeah it's that. Some weird issue crops up with making it act as a literal block so I'm forced to use raw rst syntax
